microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -168,6 +169,41 @@ def _check_rx_or_tx_pps(
                 "than 2^20 (~1m) PPS after sriov disable."
             ).is_less_than(2 ** 20)
 
+    @TestCaseMetadata(
+        description="""
+            verify vpp is able to detect azure network interfaces
+            - run fd.io vpp install scripts
+            - install vpp from their repositories
+            - start vpp service
+            - check that azure interfaces are detected by vpp
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_vpp(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+
+        vpp = DpdkVpp(node)
+        vpp.install()
+
+        net = Nics(node)
+        net.initialize()
+        _, nic = net.get_test_nic()
```

Comment:
1. how about change the method like `get_nic_by_index(index: int = -1)`. Positive number and zero means find from start, negative number means find from last.
2. I'm not clear about the first int, which is always 0. It's used in one place only. How about hardcode it in where it's used?

Reply:
That works, I'll refactor and add a second commit to this PR

Reply:
Do you have a plan to modify the method to accept an index? It may be useful in more case. And the "test_nic" is not clear from the name.

Reply:
modified it! Should be good to go